docs/src/osdk/reference/manifest.md

```diff
@@ -10,48 +10,76 @@ If there is only one crate and no workspace,
 the file is placed in the crate's root directory.
 
 For a crate inside workspace,
-it may have two related manifests,
+it may have two distinct related manifests,
 one is of the workspace
 (in the same directory as the workspace's `Cargo.toml`)
 and one of the crate
 (in the same directory as the crate's `Cargo.toml`).
-So which manifest should be used?
-
-The rules are:
-
-- If running commands in the workspace root directory,
-the `OSDK.toml` of the workspace will be used
-- If running commands in the crate directory
-  - If the crate has `OSDK.toml`,
-  the `OSDK.toml` of the crate will be used.
-  - Otherwise, the `OSDK.toml` of the workspace will be used.
+OSDK will firstly refer to the crate-level manifest, then
+query the workspace-level manifest for undefined fields.
+In other words, missing fields of the crate manifest
+will inherit values from the workspace manifest.
 
 ## Configurations
 
 Below, you will find a comprehensive version of
-the available configuration options in the manifest.
+the available configuration tree in the manifest.
 
-```toml
-kcmd_args = ["init=/bin/busybox", "path=/usr/local/bin"] # <1>
-init_args = ["sh", "-l"] # <2>
-initramfs="./build/initramfs.cpio.gz" # <3>
+Here are notes for some fields with special value treatings:
+ - `*` marks the field as "will be evaluated", that the final
+value of string `"S"` will be the output of `echo "S"` using the
+host's shell.
+ - `+` marks the path fields. The relative paths written in the
+path fields will be relative to the manifest's enclosing directory.
 
-[boot]
-loader = "grub" # <4>
-protocol = "multiboot2" # <5>
-grub-mkrescue = "/usr/bin/grub-mkrescue" # <6>
-ovmf = "/usr/bin/ovmf" # <7>
+If values are given in the tree that's the default value inferred
+if that the field is not explicitly stated.
 
-[qemu]
-path = "/usr/bin/qemu-system-x86_64" # <8>
-machine = "q35" # <9>
-args = [ # <10>
-    "-enable-kvm",
-    "-m 2G", 
-    "-device virtio-keyboard-pci,disable-legacy=on,disable-modern=off"
-] 
+```
+project_type             # The type of the current crate. Can be lib/kernel[/module]
+
+# --------------------------- the default schema settings ---------------------
+vars = []                # List of lists. These are the env vars that will be set before
```

Comment:
They are the same thing.

Maybe keep "KEY" "VALUE". Since "KEY=VALUE" need extra parsing. And whether to strip spaces between "=" needs clarification.